plugins/search-backend/src/types.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IndexableDocument } from '@backstage/plugin-search-indexer-backend';
+
+export interface SearchQuery {
+  term: string;
+  filters?: Record<string, any>;
+  pageIndex?: number;
+  pageSize?: number;
+}
+
+export interface SearchResultSet {
+  results: IndexableDocument[];
```

Comment:
It's possible we want to wrap each document up in another object to make it possible to supply things like scoring information

Reply:
Agreed. A lot of the time when one chooses a direct item, or unwrapped array, one tends to regret it later :)

Reply:
Love it. Reminds me of how `gRPC` forces you to define a message type for your service method, even if it's empty to start.